docs/cluster_api.md

```diff
@@ -0,0 +1,121 @@
+- `ObjectMetadata [metav1.ObjectMetadata]`
+- `TypeMetadata [metav1.TypeMetadata]`
+
+#### Spec
+- `MachineSetConfig [MachineSetConfig,inline]` *Configuration for the machine set (see below)*
+
+#### Status
+- `NodesProvisioned [int]` *Number of nodes provisioned by the cloud provider*
+- `NodesReady [int]` *Number of nodes ready*
+- `Conditions [array]` *Detailed status*
+  - `Type [string]` *Type of condition - Provisioned, Ready, etc*
+  - `Status [string]` *Status of the condition - True, False, Unknown*
+  - `LastProbeTime [string]` *-optional- last time the condition was checked*
+  - `LastTransitionTime [string]` *-optional- last time the condition transitioned*
+  - `Message [string]` *-optional- Human readable message associated with the condition*
+  - `Reason [string]` *-optional- Single word camel-case reason for the condition*
+
+
+### MachineSetConfig
+  - `NodeType [NodeType:string]` *Enumerated node type for the machine set (compute or master)*
+  - `Infra [boolean]` *True if this machine set should contain infrastructure pods (registry, router)*
+  - `Size [int]` *How many machines should be created for this machine set*
+  - `Version [corev1.LocalObjectReference]` *Reference to ClusterVersion resource*
+  - `Hardware`
+    - `AWS` *-optional- contains AWS specific configuration*
+      - `InstanceType [string]` *EC2 instance type (ie. m4.xlarge)*
+      - `AMIName [string]` *Name of AMI used to create this machine set (derived from Version object)*
+  - `NodeLabels [map[string]string]` *Labels to apply to nodes created in this MachineSet*
+
+### ClusterVersion
+Standalone resource that encapsulates details of a named version. It
+defines general OpenShift details about a particular version such as 
+the image template and RPM version as well as cloud-specific information
+such as the AWS AMI name.
+
+A set of predefined ClusterVersion resources can be initially created in a shared 
+namespace (like `openshift`) to define the versions supported out of the box.
+Users are able to create their own in their own namespace and associate their
+clusters with that.
+
+- `ObjectMetadata (inline) [metav1.ObjectMetadata]`
+- `TypeMetadata (inline) [metav1.TypeMetadata]`
+
+#### Spec
+- `ImageURL [string]` *-optional- template for images to be used on the cluster (oreg_url in ansible)*
+- `ImageTag [string]` *-optional- the image tag for OpenShift images (either ImageURL or ImageTag can be specified, but not both, openshift_image_tag in ansible)*
+- `RPMRepositories [array]` *-optional- description of RPM repositories to use in the cluster (openshift_additional_repos in ansible)*
+  - `URL [string]` *URL to the RPM repository*
+  - `GPGKey [string]` *URL to GPG key for repository*
+- `PackageVersion [string]` *-optional- RPM package version (openshift_pkg_version in ansible)*
+- `Hardware` *contains cloud-provider specific configuration (AWS for now)*
+  - `AWS` 
+    - `AMIName [string]` *name of the AMI that corresponds to this version*
```

Comment:
This is defined in MachineSetConfig as well as here. I get why it's defined in MachineSetConfig. Why is it defined here as well?

In fact, why is Hardware defined here at all?